wagtail_localize_rws_languagecloud/models.py

```diff
@@ -111,11 +117,41 @@ def is_failed(self):
         return self.lc_source_file_id == "" and self.create_attempts >= 3
 
     @property
-    def instance_is_published(self):
+    def published_status(self):
+        PUBLISHED = gettext_lazy("Translations published")
+        RECEIVED = gettext_lazy("Translations received")
+        READY_TO_REVIEW = gettext_lazy("Translations ready for review")
+
         instance = self.translation.get_target_instance()
+
         if not isinstance(instance, Page):
-            return True
-        return not instance.has_unpublished_changes
+            # snippets are always published immediately
+            return PUBLISHED
+
+        if not self.revision:
+            """
+            There's a couple of reasons we can end up here:
+            1. For legacy reasons: we didn't always save a revision on this object.
+               Objects created with <=0.8.0 willl have a NULL revision
+            2. If we threw an error trying to call save_target() in import_po()
+
+            In this case, say we've got the translations back
+            but we don't know if they are published or not
+            """
+            return RECEIVED
+
+        if (
```

Comment:
nitpick/suggestion: add comments before the next 2 `if`s. imho will help with the cognitive load for future us.

Took me a bit to go over the various scenarios, trying to see whether anything was missed.

So this is: "The imported file is for the latest translation revision, and the translation doesn't have any changes, so show as published"

The other being: "The imported file came from an earlier translation revision, so show as published"

not blocking, and could live as a comment here

Reply:
Yes I will add a couple of comments before I merge it (lets see if I need to tweak the logic slightly first)